hibike/devices/YogiBear/YogiBear.cpp

```diff
@@ -1,122 +1,258 @@
+      return sizeof(float);
+      break;
+
+    case PID_POS_SETPOINT: 
+      setPosSetpoint(((float *)data)[0]);
+      return sizeof(float);
+      break;
+
+    case PID_POS_KP: 
+      setPosKP(((float *)data)[0]);
+      return sizeof(float);
+      break;
+
+    case PID_POS_KI: 
+      setPosKI(((float *)data)[0]);
+      return sizeof(float);
+      break;
+
+    case PID_POS_KD: 
+      setPosKD(((float *)data)[0]);
+      return sizeof(float);
+      break;
+
+    case PID_VEL_SETPOINT: 
+      setVelSetpoint(((float *)data)[0]);
+      return sizeof(float);
+      break;
+
+    case PID_VEL_KP: 
+      setVelKP(((float *)data)[0]);
+      return sizeof(float);
+      break;
+
+    case PID_VEL_KI: 
+      setVelKI(((float *)data)[0]);
+      return sizeof(float);
+      break;
+
+    case PID_VEL_KD: 
+      setVelKD(((float *)data)[0]);
+      return sizeof(float);
+      break;
+
+    case CURRENT_THRESH: 
+      setCurrentThreshold(((float *)data)[0]);
+      return sizeof(float);
+      break;
 
-    case DUTY:
-      if (value <= 100) && (value >= 0) {
-        duty = value;
+    case ENC_POS: 
+      if((float) data[0] == 0) {
+        zeroEncoder();
+        return sizeof(float);
       }
-      return duty;
+
       break;
-      
-    case FORWARD:
-      forward = 1;
-      inputA = 0;
-      inputB = 1;
-      return forward;
+
+    case ENC_VEL: 
       break;
 
-    case REVERSE:
-      forward = 0;
-      inputA = 1;
-      inputB = 0;
-      return forward;
+    case MOTOR_CURRENT: 
       break;
 
     default:
-      return ~((uint32_t) 0);
+      return 0;
   }
+
+  return 0;
 }
 
-// you must implement this function. It is called when the device receives a DeviceStatus packet.
-// the return value is the value field of the DeviceRespond packet hibike will respond with
-uint32_t device_status(uint8_t param) {
-  // if (param < NUM_PARAMS) {
-  //   return params[param];
-  // }
+
+// You must implement this function.
+// It is called when the device receives a Device Data Update packet.
+// Modifies data_update_buf to contain the parameter value.
+//    param           -   Parameter index
+//    data_update_buf -   buffer to return data in, little-endian
+//    buf_len         -   Maximum length of the buffer
+//
+//    return          -   sizeof(value) on success; 0 otherwise
+
+uint8_t device_read(uint8_t param, uint8_t* data_update_buf, size_t buf_len) {
+  float* float_buf;
+
   switch (param) {
-    case DUTY:
-      return duty;
+    case ENABLE:
+      data_update_buf[0] = readMotorEnabled();
+      return sizeof(bool);
+      break;
+
+    case COMMAND_STATE: 
+      data_update_buf[0] = driveMode;
+      return sizeof(uint8_t);
+      break;
+
+    case DUTY_CYCLE: 
+      float_buf = (float *) data_update_buf;
+      float_buf[0] = readPWMInput();
+      return sizeof(float);
+      break;
+
+    case PID_POS_SETPOINT: 
+     break;
+
+    case PID_POS_KP: 
+      break;
```

Comment:
you can stack all of these case statements and just have one `return 0` / `break` for all of them